catalyst_voices/lib/pages/workspace/proposal_details.dart

```diff
@@ -121,43 +120,50 @@ class _StepDetails extends StatelessWidget {
   const _StepDetails({
     super.key,
     required this.id,
-    required this.name,
+    required this.title,
+    this.titleInDetails,
     this.desc,
-    this.doc,
+    this.richTextParams,
     this.isSelected = false,
     this.isEditable = false,
   });
 
   final int id;
-  final String name;
+  final String title;
+  final String? titleInDetails;
   final String? desc;
-  final Document? doc;
+  final RichTextParams? richTextParams;
   final bool isSelected;
   final bool isEditable;
 
   @override
   Widget build(BuildContext context) {
-    return (desc != null)
-        ? WorkspaceTextTileContainer(
-            name: name,
-            isSelected: isSelected,
-            headerActions: [
-              TextButton(
-                onPressed: isEditable ? () {} : null,
-                child: Text(
-                  context.l10n.stepEdit,
-                  style: Theme.of(context).textTheme.labelSmall,
-                ),
-              ),
-            ],
-            content: desc!,
-          )
-        : WorkspaceTileContainer(
-            isSelected: isSelected,
-            content: VoicesRichText(
-              title: name,
-              document: doc,
+    if (desc != null) {
+      return WorkspaceTextTileContainer(
+        name: titleInDetails != null ? titleInDetails! : title,
```

Comment:
if `titleInDetails` overrides the `title`, can we just set `title` to whatever is in `titleInDetails`?

Reply:
I don't understand what do you mean.
Do you know what `titleInDetails` is for?

<img width="1093" alt="Screenshot 2024-10-25 at 13 28 04" src="https://github.com/user-attachments/assets/1ca3966b-5b1d-4dff-b9b3-51bcb44b0d6c">

Reply:
I was suggesting to simplify the logic by extracting the decision whether to use the `title` or `titleInDetails` to the parent widget. `_StepDetails` could just have a single parameter called title, and you would feed it like:

```
      return _StepDetails(
                key: ValueKey('WorkspaceStep${step.id}TileKey'),
                id: step.id,
                title: step.titleInDetails ?? step.title,
                desc: step.description,
                richTextParams: step.richTextParams,
                isSelected: step.id == selected,
                isEditable: step.isEditable,
              );
```

This way we don't have this condition repeated: `titleInDetails != null ? titleInDetails! : title,`.